WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/PointOfSaleItemListErrorView.swift

```diff
@@ -0,0 +1,66 @@
+import SwiftUI
+
+struct PointOfSaleItemListErrorView: View {
+    private var viewModel: any ItemListViewModelProtocol
+
+    init(viewModel: any ItemListViewModelProtocol) {
+        self.viewModel = viewModel
+    }
+
+    var errorContents: ItemListViewModel.ErrorModel {
+        guard let errorContents = viewModel.state.hasError else {
+            return ItemListViewModel.ErrorModel(title: "Unknown error",
+                                                subtitle: "Unknown error",
+                                                buttonText: "Retry")
+        }
+        return errorContents
+    }
+
+    var body: some View {
+        VStack(alignment: .center, spacing: Constants.headerSpacing) {
+            Text(Localization.productTitle)
+                .frame(maxWidth: .infinity, alignment: .leading)
+                .padding(.vertical, Constants.headerPadding)
+                .font(Constants.titleFont)
+                .foregroundColor(Color.posPrimaryTexti3)
+            Spacer()
+            POSErrorExclamationMark()
+            Text(errorContents.title)
+                .foregroundStyle(Color.posPrimaryTexti3)
+                .font(.posTitle)
+                .bold()
+            Divider()
+                .padding([.leading, .trailing], Constants.buttonPadding)
+            Text(errorContents.subtitle)
+                .foregroundStyle(Color.posPrimaryTexti3)
+                .font(.posBody)
+                .padding([.leading, .trailing])
+            Button(action: {
+                Task {
+                    await viewModel.reload()
+                }
+            }, label: {
+                Text(errorContents.buttonText)
+            })
+            .buttonStyle(POSPrimaryButtonStyle())
+            .padding([.leading, .trailing], Constants.buttonPadding)
+            Spacer()
+        }
+    }
+}
+
+private extension PointOfSaleItemListErrorView {
+    enum Localization {
+        static let productTitle = NSLocalizedString(
+            "pos.pointOfSaleItemListErrorView.productTitle",
+            value: "Products",
+            comment: "Title for the Point of Sale screen"
+        )
+    }
+    enum Constants {
+        static let headerPadding: CGFloat = 8
+        static let headerSpacing: CGFloat = 16
+        static let titleFont: Font = .system(size: 40, weight: .bold, design: .default)
+        static let buttonPadding: CGFloat = 240
```

Comment:
Nitpick;

Technically it's true, that we have a 240-point margin. However, usually when we have such a big margin the element sizing follows a different logic. For example, the min width of a button is half of the screen-size width.

Reply:
Right! I updated this on 2ce4c6fe3c54dd34e66835cb9d1a7cfbe8d1a659 by setting a specific width for the button. My attempts using minWidth were defaulting to `.infinity` for some reason, despite setting a max too 🤔 

Another option that works is setting it specifically via `.frame(width: UIScreen.main.bounds.width / 2)` but I'm not quite sure if that's a good approach if we're purely using SwiftUI? Thoughts?

Reply:
I think the best is to use `Geometry` to set the proportional width. However, I think had some issues myself with that so chose to stick with `(UIScreen.main.bounds.width / 2)`.